src/psyclone/core/access_info.py

```diff
@@ -100,21 +96,55 @@ def change_read_to_write(self):
         self._access_type = AccessType.WRITE
 
     @property
-    def indices(self):
+    def indices_groups(self):
         '''
-        :returns: the indices used in this access. Can be None.
-        :rtype: list of :py:class:`psyclone.psyir.nodes.Node`, or None
+        This function returns the list of accesses used for each component,
+        e.g. `a(i)%b(j,k)%c` will return `[ [i], [j, k], [] ]`. Especially
+        in case of a simple scalar variable `a` this function will still
+        return `[ [] ]`. Each entry in this list of lists will be a PSyIR
```

Comment:
"...will be the PSyIR for the index expression used."

Reply:
Done.